circuits/plonk/src/scalars.rs

```diff
@@ -69,3 +67,126 @@ impl<F: Field> RandomOracles<F> {
         }
     }
 }
+
+//
+// OCaml types
+//
+
+#[cfg(feature = "ocaml_types")]
+pub mod caml {
+    use super::*;
+    use ocaml_gen::OcamlGen;
+    use oracle::sponge::caml::CamlScalarChallenge;
+
+    //
+    // ProofEvaluations<F> <-> CamlProofEvaluations<CamlF>
+    //
+
+    #[derive(Clone, ocaml::IntoValue, ocaml::FromValue, OcamlGen)]
```

Comment:
Can you check the generated assembly code and confirm that we're not doing extra 'deep deconstruct-reconstruct' clones due to this? If we are, lets just make the type parameter represent `'f array` instead of `'f` and deal with this on the OCaml end.

Reply:
I think you're catching a potential performance regression from this PR: https://github.com/o1-labs/proof-systems/pull/121/

IIRC, I didn't get why the types were not clearly defined if they were always vectors. 

I can create an issue for this (as this seems unrelated to this PR) if that's OK? (Also I'm a bit worried about  handling `Vec<F>` as a generic type parameter within ocaml-gen.)